department-of-veterans-affairs/caseflow
# Shalliburton/13688 validate document

Resolves [13688](https://github.com/department-of-veterans-affairs/caseflow/issues/13688)

### Description
Moved document ID validator from update legacy appeal workflow into a shared module and included in the updated legacy appeal workflow as well as Case Assignment. VACOLS case reviews can now use the same logic to check for a valid document ID as the legacy appeals.

### Acceptance Criteria
- [x] Code compiles correctly and tests pass
- [x] Acting Judge has one legacy appeal with Judge Task and one legacy appeal with Attorney Task

### Testing Plan
**NOTE:** You will need to reseed your DB to pick up the new legacy appeals
1. Become Acting Judge Kris
2. Navigate to the user's queue and verify there are 2 legacy appeals
3. Navigate to the first appeal and check the Document ID. 
- If it is NONE, verify the actions are related to an Attorney task. 
- If it contains an ID, verify that the actions are related to a Judge Task.
4. Navigate to the second appeal and repeat the steps above for the other Document ID
 
### Documentation Updates
- [x] Added description on seeding Case Reviews in FACOLS to [wiki](https://github.com/department-of-veterans-affairs/caseflow/wiki/FACOLS)

## Discovery Notes
https://hackmd.io/DsIbAObHRoeL7K_9BqqojA
